data_structures/bloom_filter.cpp

```diff
@@ -0,0 +1,280 @@
+/**
+ * @file
+ * @brief [Bloom Filter](https://en.wikipedia.org/wiki/Bloom_filter)
+ * generic implementation in C++
+ * @details A Bloom filter is a space-efficient probabilistic data structure,
+ * a query returns either "possibly in set" or "definitely not in set".
+ *
+ * More generally, fewer than 10 bits per element are required for a 1% false
+ * positive probability, independent of the size or number of elements in the
+ * set.
+ *
+ * It helps us to not make an "expensive operations", like disk IO - we can
+ * use bloom filter to check incoming request, and with a good probability
+ * get an answer of bloom filter, that we don't need to make our "expensive
+ * operation"
+ *
+ *
+ * [Very good use case example](https://stackoverflow.com/a/30247022)
+ *
+ * Basic bloom filter doesn't support deleting of elements, so
+ * we don't need to implement deletion in bloom filter and bitset in our case.
+ * @author [DanArmor](https://github.com/DanArmor)
+ */
+
+#include <cassert>           ///< for assert
+#include <functional>        ///< for list of hash functions for bloom filter constructor
+#include <initializer_list>  ///< for initializer_list for bloom filter constructor
+#include <string>            ///< for testing on strings
+#include <vector>            ///< for vector
+
+/**
```

Comment:
You might have to update the functions/classes to start with `data_structures::`.
```suggestion
/**
 * @namespace data_structures
 * @brief Data Structures algorithms
 */
namespace data_structures {
/**
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Tags:** Inefficient

Reply:
Done